src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -713,9 +716,14 @@ private void AddEasyHandle(RequestCompletionSource state)
                         throw new HttpRequestException(SR.net_http_client_execution_error,
                             GetCurlException(result, true));
                     }
+                    state.SessionHandle = _multiHandle;
+                    if (null == _multiHandle.FdSet)
+                    {
+                        _multiHandle.FdSet = new HashSet<int>();
+                        PollFunction(_multiHandle);
```

Comment:
In the code I was commenting on, you were only calling PollFunction if _multiHandle.FdSet was null, which would only happen the first time it's called, hence my question.  I was instead expecting this to be something more like:
```C#
if (_multiHandle.FdSet == null)
    _multiHandle.FdSet = new HashSet<int>();
if (_multiHandle.FdSet.Count == 0)
    PollFunction(_multiHandle);
```
I see that you've addressed that in the most recent commits you pushed, so thanks.